packages/pages/src/upgrade/upgrade.ts

```diff
@@ -1,20 +1,28 @@
 import { Command } from "commander";
-import path from "path";
 import { updatePages } from "./pages_updater.js";
 import { migrateConfigs } from "./migrate_config.js";
-import { execSync } from "child_process";
+import { ProjectStructure } from "../common/src/project/structure.js";
+import { templatesHandler } from "../generate/templates/templates.js";
+import { artifactsHandler } from "../generate/artifacts/artifacts.js";
+import { cleanConfigs } from "./clean_configs.js";
 
-const handler = async () => {
-  const rootPath = path.resolve("");
-  await updatePages(rootPath);
-  await migrateConfigs(rootPath);
-  execSync("pages generate templates");
-  execSync("pages generate artifacts");
+const handler = async ({ scope }: { scope: string }) => {
+  const scoped = { scope: scope };
+  const projectStructure = await ProjectStructure.init(scoped);
+  await updatePages(projectStructure);
+  await migrateConfigs(projectStructure);
+  await templatesHandler(scoped);
```

Comment:
Can you not pass scope directly through?

Reply:
We could, but they would all be init-ing the project structure on the other end anyways

Reply:
I meant can you pass scope instead of scoped

Reply:
oh it doesn't accept that

It expects {scope: scope}, so I could write that multiple times or pass the object through

Reply:
Cleaned this up a bit. We pass {scope: scope} through